### PR TITLE
feat(security): Support multiple applications in `ApplicationNameable`

### PR DESCRIPTION
**Pull request by ajordens**
Allow an implementation of `ApplicationNameable` to support authorization
checks against _multiple_ applications.

This handles scenarios where a particular operation may target multiple
server groups that may or may not exist in the same application.

### CODE REVIEW

**Comment by ajordens** [comment]
I err'd on the side of using `ApplicationNameable` and `ServerGroupNameable` since they were more descriptive of the purpose and commonly used. It didn't strike me as abnormal for these to be plural as we do have operations that work on more than one resource.

The alternative would have been to do some conversion from `ServerGroupNameable` usage to `ResourcesNameable` since it already offered a plural `getApplications()`.

I will be going through all aws and titus descriptions and implementing the appropriate interfaces as we're fairly wide open with respect to application restriction.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/BasicAmazonDeployDescription.groovy [hunk 1]
```diff
@@ -20,12 +20,13 @@ import com.netflix.spinnaker.clouddriver.aws.model.AmazonAsgLifecycleHook
...
 import com.netflix.spinnaker.clouddriver.deploy.DeployDescription
 import com.netflix.spinnaker.clouddriver.orchestration.events.OperationEvent
+import com.netflix.spinnaker.clouddriver.security.resources.ApplicationNameable
 import groovy.transform.AutoClone
 import groovy.transform.Canonical
 
 @AutoClone
 @Canonical
-class BasicAmazonDeployDescription extends AbstractAmazonCredentialsDescription implements DeployDescription {
+class BasicAmazonDeployDescription extends AbstractAmazonCredentialsDescription implements DeployDescription, ApplicationNameable {
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/BasicAmazonDeployDescription.groovy]: Subsequently I might pull the `implements ApplicationNameable` up to `DeployDescription` and all cloud providers will pick it up.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/ResizeAsgDescription.groovy [hunk 2]
```diff
@@ -17,18 +17,16 @@
...
 
-class ResizeAsgDescription extends AbstractAmazonCredentialsDescription {
-  String serverGroupName
-  String region
+class ResizeAsgDescription extends AbstractAmazonCredentialsDescription implements ServerGroupNameable {
   ServerGroup.Capacity capacity
   List<AsgTargetDescription> asgs = []
 
-  @Deprecated
-  String asgName
```
Comment (ajordens) [inline clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/ResizeAsgDescription.groovy]: Pretty sure these aren't necessary anymore ... we've got a `DescriptionPreProcessor` that takes all these `@Deprecated` values and places them consistently under `asgs`.

File: clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/BasicAmazonDeployDescription.groovy (hunk 1 above)
Comment (cfieber) [reply clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/description/BasicAmazonDeployDescription.groovy]: seems reasonable to me

**Pull request merged**